kernel/kernel-component/src/main/webapp/WEB-INF/user-components.xml

```diff
@@ -116,4 +116,22 @@
 		<property name="memoryService" ref="org.sakaiproject.memory.api.MemoryService" />
 	</bean>
 
+	<bean id="org.sakaiproject.user.api.CandidateDetailProvider"
```

Comment:
Sorry but, 
we don't understand why do you say this will fail if the kernel is loaded after. Providers does not have any service related with that, so it's not possible another service overrides the kernel definition.

It's true, at the beginning, we had some doubts about the kernel getting loaded before the providers. Because the providers component contains the related implementations and we didn't know if the classloader could read them. Anyway, at this moment this problem seems ok.

So please, can you explain a bit more the problem and if you have any posible solution?
Thanks